src/kiota/Rpc/PluginResultMapper.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Microsoft.DeclarativeAgents.Manifest;
+using Microsoft.OpenApi.Extensions;
+
+namespace kiota.Rpc
+{
+    internal class PluginResultMapper
+    {
+        private static string? GetReferenceId(Auth? auth)
+        {
+            if (auth == null)
+            {
+                return null;
+            }
+            else if (auth is OAuthPluginVault)
+            {
+                return (auth as OAuthPluginVault)?.ReferenceId;
+            }
+            else if (auth is ApiKeyPluginVault)
+            {
+                return (auth as ApiKeyPluginVault)?.ReferenceId;
+            }
+            else
+            {
+                return null;
+            }
```

Comment:
```suggestion
            return null;
```

Reply:
Solved in further refactor of the method